app/src/main/java/com/example/jeff/newmovie/MainMovie.java

```
@@ -0,0 +1,144 @@
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+
+public class MainMovie extends AppCompatActivity {
+
+    ArrayList<Movie> arrayListMovies = new ArrayList<Movie>();
+    GridView gvMovies;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.main_movie_activity);
+        gvMovies = (GridView)findViewById(R.id.gvMovie);
+        new FetchMovieData().execute("http://api.themoviedb.org/3/discover/movie?api_key=c8e968bcd558585cd3f858dda44ae89d");
+
+    }
+
+
+
+
+    public class FetchMovieData extends AsyncTask<String,Void,String>{
+
+        private  String LOG_TAG = "MOVIE JSON";
+        private String jsonData;
+        private final String IMAGE_URL = "http://image.tmdb.org/t/p/w300/";
+
+        @Override
+        protected String doInBackground(String... params) {
+
+
+            BufferedReader reader = null;
+            HttpURLConnection connection = null;
+
+            try {
+                URL url = new URL(params[0]);
+                connection = (HttpURLConnection) url.openConnection();
+                connection.setRequestMethod("GET");
+                connection.connect();
+
+
+
+            } catch (MalformedURLException e) {
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
+            } finally {
+                if (connection == null){
+                    connection.disconnect();
+                }
+            }
+            InputStream stream = null;
+            try {
+                stream = connection.getInputStream();
+            } catch (IOException e) {
+                e.printStackTrace();
```

Dapat propagate mo yun exception (throw e) instead na swallow.